### PR TITLE
react/states_and_effects: add cv application project lesson

### PR DESCRIPTION
**Pull request by Lofty-Brambles**
## Because

It'd revamp the existing CV application project.

## This PR

- Adds a new lesson under react/states_and_effects

## Issue

Closes https://github.com/TheOdinProject/top-meta/issues/265

## Additional Information

N/A

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/states_and_effects/project_cv_application.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+### Introduction
+
+Congratulations, you made it to the very first React project. In this project, you will be creating a small CV application. It's a great project to get you started because it includes many of the basic concepts you've learned so far, without a huge overhead of complexity.
+
+### Deploying A React App
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:5]:
```suggestion
### Deploying a React application
```

File: react/states_and_effects/project_cv_application.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+While we have been using GitHub Pages to deploy our **static webpages** till now, and we can definitely continue to do so with hacky ways, it's far easier to let a PaaS (Platform as a Service) do the same for us! There are a lot of options to host such sites, and the [Vite hosting docs](https://vitejs.dev/guide/static-deploy.html) go over all of them.
+
+Not only do some of these get rid of additional steps after pushing our code to GitHub as these services have tools that automatically _deploy on push_, but you also will not need to worry about other issues further down the curriculum, like routers and build steps. We shall use a few selected options for now. Once you are done with the assignment, choose any of the following options, and deploy!
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:9]:
```suggestion
Not only do some of these get rid of additional steps after pushing our code to GitHub as these services have tools that automatically _deploy on push_, but you also don't have to worry about other issues further down the curriculum, like routers and build steps. We shall use a few selected options for now. Once you are done with the assignment, choose one of the following options, and deploy!
```

File: react/states_and_effects/project_cv_application.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+---
+
+### Netlify
+
+While there are many ways to deploy to Netlify, like uploading your `dist` directly or using [Netlify's `netlify-cli`](https://docs.netlify.com/cli/get-started/) CLI, the most convenient way would be to directly import your GitHub repository to Netlify.
+
+1.  Push your React app to GitHub.
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:17]:
```suggestion
1.  Push your React application to GitHub.
```

File: react/states_and_effects/project_cv_application.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+- [Documentation Website](https://docs.netlify.com/)
+- [This section of the Vite docs](https://vitejs.dev/guide/static-deploy.html#netlify-with-git) has a similar brief guide for deployment
+
+---
+
+### Vercel
+
+Similar to Netlify, Vercel also offers a few ways to deploy, but we will be importing our GitHub repository to get benefits like deploy-on-push.
+
+1.  Again, push your React app to GitHub.
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:34]:
```suggestion
1.  Again, push your React application to GitHub.
```

File: react/states_and_effects/project_cv_application.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+#### Links
+
+- [Homepage](https://www.vercel.com/)
+- [Documentation Website](https://www.vercel.com/docs)
+
+---
+
+### Cloudflare Pages
+
+1.  Yes, push your React app to GitHub.
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:47]:
```suggestion
1.  Yes, push your React application to GitHub.
```

File: react/states_and_effects/project_cv_application.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+- [Homepage](https://pages.cloudflare.com/)
+- [Documentation Website](https://developers.cloudflare.com/pages)
+- [This section in the Cloudflare docs](https://developers.cloudflare.com/pages/framework-guides/deploy-a-vite3-project/) discusses these very steps
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  Create a new React Project.
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:65]:
```suggestion
1.  Create a new React project.
```

File: react/states_and_effects/project_cv_application.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+2.  Think about how to structure your application into components. Your application should include:
+    1.  A section to add general information like name, email and phone number.
+    2.  A section to add your educational experience (school name, title of study and date of study)
+    3.  A section to add practical experience (company name, position title, main tasks of your jobs, date from and until when you worked for that company)
```
Comment (fcasibu) [inline react/states_and_effects/project_cv_application.md:69]:
```suggestion
    3.  A section to add practical experience (company name, position title, main responsibilities of your jobs, date from and until when you worked for that company)
```

**Review by fcasibu — COMMENTED** [review]
Great job with the lesson, love the options as well 🚀 Just added a few comments

**Review by 01zulfi — CHANGES_REQUESTED** [review]
Can we put the `Deploying a react app` section after the `Assignment` section please? Also there would be minor changes needed accordingly. The reasoning is that I wouldn't want learners to worry about deploying until they've actually completed the project, hope it makes sense

**Comment by Lofty-Brambles** [comment]
@01zulfi - made the changes

**Review by 01zulfi — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**